### DISCUSSION TITLE
에뮬레이터 (LD플레이어) 에서 디버깅 하는 방법 (adb logcat)

### DISCUSSION THREAD

**Discussion by smilejsu82** [naverz/zepeto-studio-kor#641] [3. Creator Samples]
LD플레이어를 설치 합니다 

그리고 LD Store에서 제페토를 검색해 설치 해주세요 
![image](https://user-images.githubusercontent.com/50188795/164143707-4c5a987a-6eca-4623-aa6d-c06996c1c33a.png)

설정에 들어갑니다 
![image](https://user-images.githubusercontent.com/50188795/164143841-5360d6e6-0420-4368-932e-baaf846e8433.png)

기타에 들어가서  ADB 디버깅과 Root권한을 설정해주세요 
![image](https://user-images.githubusercontent.com/50188795/164143885-6bd2546f-6af6-4df3-b131-9c5b43bee0d9.png)

이제 유니티 Preferences 창을 열고 Android SDK 경로를 복사 해서 해당 폴더로 이동 합니다 
그리고 adb.exe 파일이 있는지 확인해요 
![image](https://user-images.githubusercontent.com/50188795/164144010-0a799041-fba8-47df-8785-decd198278d8.png)
![image](https://user-images.githubusercontent.com/50188795/164144048-45e2547b-c7a3-43ef-b2ce-972603d6b210.png)

cmd 창을 열고 
해당 경로로 이동해요 
![image](https://user-images.githubusercontent.com/50188795/164144125-fd40b361-b393-4467-88ea-233be0a8220d.png)

adb devices 명령어를 입력하면 LD플레이어가 보입니다 
![image](https://user-images.githubusercontent.com/50188795/164144200-6ccf4054-5986-423c-a00c-1cd58e8463f2.png)

유니티에서 제페토 버튼을 눌러 QR코드를 만들어 줍니다 
![image](https://user-images.githubusercontent.com/50188795/164145766-851349e1-82bb-4250-8f97-14758547922a.png)
![image](https://user-images.githubusercontent.com/50188795/164146005-666e29db-f706-4a0d-8813-758653c7a376.png)

캠이 있으면 LD플레이어에서 QR코드 리더기로 읽으면 대지만 없다면 QR코드를 (시프트+ 윈도우 + S)키를 눌러 화면을 캡쳐 합니다 
그림판에 붙여 넣고 저장 해요 
![image](https://user-images.githubusercontent.com/50188795/164146102-7e94cd9a-e074-4f9c-9bae-a453f143d4c7.png)
![image](https://user-images.githubusercontent.com/50188795/164146129-212c48ee-6560-4d4b-bb3b-87931c74cc97.png)

그리고 QR코드 이미지를 끌어다가 LD플레이어에 놓습니다 
![image](https://user-images.githubusercontent.com/50188795/164146173-def1e9a3-b091-41fc-8ab5-9c48e2349ca3.png)

LD플레이어에서 브라우저를 열고 다음 주소를 입력 해요 
[https://www.onlinebarcodereader.com/](url)

![image](https://user-images.githubusercontent.com/50188795/164145139-dee7f9a9-26a0-4c1e-8cde-5fac00ae995d.png)

upload 파일을 선택해 갤러리에서 아까 저장한 QR코드 이미지를 선택 해요 
![image](https://user-images.githubusercontent.com/50188795/164146246-bc4e017b-73e1-46f5-a06a-ec1aeca37bc8.png)
![image](https://user-images.githubusercontent.com/50188795/164146214-5d3f5eaf-003d-4e4c-aafc-ced9b3d0c29c.png)
![image](https://user-images.githubusercontent.com/50188795/164146271-db1c6c07-9ca5-4936-9134-fdd144372378.png)

그리고 START버튼을 눌러 줍니다 
![image](https://user-images.githubusercontent.com/50188795/164146322-3b2e7606-71db-45ea-abea-4321049d4497.png)

접속 링크가 나면 클릭합니다 
![image](https://user-images.githubusercontent.com/50188795/164146358-898e7acf-a97f-4d9c-9609-482d2b71e01d.png)

이제 월드에 접속 할거예요 
![image](https://user-images.githubusercontent.com/50188795/164146418-ee337d9d-e942-4501-ab81-e602e1635bb2.png)

cmd창으로 돌아가서 adb logcat -s Unity를 입력 하면 로그를 확인 할수 있습니다 
![image](https://user-images.githubusercontent.com/50188795/164146449-b75758f5-d55b-430f-892b-72e349cd5d2d.png)

![image](https://user-images.githubusercontent.com/50188795/164146741-b88f98f9-8dcd-455d-a0b2-58a26157e5b9.png)

![image](https://user-images.githubusercontent.com/50188795/164146543-ab61ac74-719d-4a2f-a017-1c35dfa1cc99.png)

디버깅을 중지 하려면 컨트롤 + C를 눌러줘요 
![image](https://user-images.githubusercontent.com/50188795/164146575-64a0a89b-2baf-4d20-8309-30ee7cb17d56.png)

_캠이 있다면 QR코드를 저장하고 온라인바코드리더를 통해 하는 작업을 생략 할수 있습니다_ 

개발 하시는데 조금이나마 도움이 되었으면 해요 ~ 화이팅 !! 👍

**Comment by x4cf** [comment]
와...우! 대단해요!

**Comment by smilejsu82** [comment]
> 와...우! 대단해요!

맨날 질문만하기 미안하드라구요 ㅋㅋ ^^;;

**Comment by Zepeto-TechEvangelist** [comment]
유용한 팁은 Creator Samples 카테고리로 옮겨드리겠습니다. 감사합니다 😄

**Comment by Zepeto-TechEvangelist** [comment]
![164146005-666e29db-f706-4a0d-8813-758653c7a376](https://user-images.githubusercontent.com/100118650/164359856-2af6624b-c4e2-4be2-93ab-cfb4a16bd818.png)

참고로 ZEPETO QR코드 창에 있는 Copy URL 버튼을 누르면 링크를 손 쉽게 얻어올 수 있습니다 👍

**Reply by smilejsu82** [reply]
![image](https://user-images.githubusercontent.com/50188795/164360217-bab9e69a-ca51-47d0-9d64-c1aa7b77c8e6.png)

에뮬 (LD플레이어) 에서는 안되드라구요 ^^

**Reply by smilejsu82** [reply]
![image](https://user-images.githubusercontent.com/50188795/164360307-85478f7f-ff4f-468c-90d4-95dfbc8ee728.png)
![image](https://user-images.githubusercontent.com/50188795/164360334-8c50a9a4-670c-48b1-be71-b0c330f11e4b.png)
온라인바코드리더로 url도 안되는 부분입니다 

그래서 불편 하지만 QR을 복사해서 그림판에 옮기고 하는 작업이 필요 해요 ㅠ

**Reply by x4cf** [reply]
애뮬에서 스킴 앱을 설치하고 URL링크를 넣는 방법도 가능할까요? :D

**Reply by smilejsu82** [reply]
> 애뮬에서 스킴 앱을 설치하고 URL링크를 넣는 방법도 가능할까요? :D

스킴 앱이 무엇일까요?~

**Reply by x4cf** [reply]
안드로이드 환경에서는 테스트 링크로 바로 연결되지 않아서 별도의 앱의 도움을 받아 테스트해요, 이런 앱의 도움으로 링크를 열 수 있지 않을까 싶어서요 :D
https://github.com/naverz/zepeto-studio-kor/discussions/155

**Reply by smilejsu82** [reply]
> 애뮬에서 스킴 앱을 설치하고 URL링크를 넣는 방법도 가능할까요? :D

wow 👍

**Reply by Andwhy81kr** [reply]
저는 QR코드 스캔 기능을 이용합니다.
LD플레이어에서 네이버앱 이나 제페토 앱에서 QR 코드 스캔기능을 키면 LD플레이어가 화면을 캡쳐할수 있게 기능을 제공해줘서 별도의 카메라 없이 떠있는 QR코드를 스캔해서 테스트 하고 있습니다.
https://youtu.be/QcsTF8H3iz0?t=901

15분정도쯤에 보시면 스캔하는 장면이 보이네요.
(제페토 QR코드는 LD플레이어에서 오동작하는경우가 종종있어서 Naver앱을 주로 써요)

**Reply by smilejsu82** [reply]
> 저는 QR코드 스캔 기능을 이용합니다.
> 
> LD플레이어에서 네이버앱 이나 제페토 앱에서 QR 코드 스캔기능을 키면 LD플레이어가 화면을 캡쳐할수 있게 기능을 제공해줘서 별도의 카메라 없이 떠있는 QR코드를 스캔해서 테스트 하고 있습니다.
> 
> https://youtu.be/QcsTF8H3iz0?t=901
> 
> 
> 
> 15분정도쯤에 보시면 스캔하는 장면이 보이네요.
> 
> (제페토 QR코드는 LD플레이어에서 오동작하는경우가 종종있어서 Naver앱을 주로 써요)

오오 좋은 방법이네요 참고할게요~
